### DISCUSSION TITLE
Slider "confirmed" event?

### DISCUSSION THREAD

**Discussion by lukors** [emilk/egui#1249] [Q&A] [answered]
I've recently started using this crate. I'm interested in detecting when the user has "confirmed" the value they want on a slider. So when the user does one of the following‥

1. Releases the mouse button after grabbing the slider
1. Releases the mouse button after grabbing the text field next to the slider
1. Presses enter or tab, or clicks outside the text field next to the slider while the text field is active

- I've been able to get an event for number 1 via `drag_released()`.
- I thought maybe `lost_focus()` would work for number 3, but seems like it doesn't do anything.
- it seems like dragging on the text box does not generate any drag events like dragging on the slider does.

So I can't get an event for number 2 or 3. I can of course get `changed()` events for whenever the value changes, but I'm specifically interested in knowing when the user has "settled" on a value.

My use case is handling undo/redo properly in my application. I don't want to generate an undo/redo event on every change, because that's just too much to `Ctrl` `Z` through. I can check how much time has passed and skip adding stuff to the undo queue based on that, but that would be a pretty complicated and unpredictable system that I would prefer to not have to write.

Anyone know if there's a way to do this, or should I create an issue?

**Comment by emilk** [comment, marked as answer]
> * I thought maybe `lost_focus()` would work for number 3, but seems like it doesn't do anything.

Sounds like a bug we should fix!

> * it seems like dragging on the text box does not generate any drag events like dragging on the slider does.

Let's fix that too!

Then `response.drag_released() || response.lost_focus()` should cover your needs, correct?

**Reply by lukors** [reply]
Yes, these fixes would allow me to use the slider widget! :)
Should I create issues for this, or do you want to?

**Reply by emilk** [reply]
please do!

**Reply by lukors** [reply]
Submitted bug reports: #1268 #1269
